synapse/lib/remcycle.py

```diff
@@ -0,0 +1,876 @@
+            - The second object should be a dictionary of optional values and
+              their defaults.  A user may provide alternative values when
+              calling build_http_request, but sensible defaults should be
+              provided here.
+        * http: A dictionary of key/value items which can provide per-api
+          specific arguements for the creation of HTTPRequest objects.
+        * ingests: A dictionary of Synapse ingest definitions which will be
+          used to create Ingest objects.  During registration of a Nyx object
+          with Hypnos, these will be registered into the Hypnos cortex.
+          Multiple named ingests may be made available for a single API.
+        * vars: This is a list of 2 value items which are stamped into the url
+          value during the construction of the Nyx object.
+
+    See a complete example below:
+
+    ::
+
+        {
+          "api": [
+            [
+              "someplace"
+            ],
+            {
+              "domore": 0
+            }
+          ],
+          "doc": "api example",
+          "http": {
+            "headers": {
+              "token-goodness": "sekrit token"
+            }
+          },
+          "ingests": {
+            "ingest_definition": {
+              "ingest": {
+                "forms": [
+                  [
+                    "inet:ipv4",
+                    {
+                      "var": "ip"
+                    }
+                  ]
+                ],
+                "vars": [
+                  [
+                    "ip",
+                    {
+                      "path": "ip"
+                    }
+                  ]
+                ]
+              }
+            }
+          },
+          "url": "http://vertex.link/api/v4/geoloc/{{someplace}}/info?domore={{domore}}&apikey={APIKEY}",
+          "vars": [
+            ["APIKEY", "8675309"]
+          ]
+        }
+
+    ::
+
+    This example should be interpreted as the following:
+
+        1. The APIKEY value in the 'vars' will be set in the URL, resulting in
+           the following default url:
+
+        ::
+
+            "http://vertex.link/api/v4/geoloc/{someplace}/info?domore={domore}&apikey=8675309"
+
+        ::
+
+        2. The HTTP request will have the header "token-goodness" set to
+           "sekrit token" for the request.
+        3. The caller must provide the "someplace" value in the api_args when
+           calling build_http_request.  The caller may provide the "domore"
+           value if they want to override the default value of "0".
+        4. An Ingest will be created for parsing the data from the API and
+           made available to the Hypnos object.
+
+    :param api_config: API Endpoint configuration outlined above.
+    :param namespace_http_config: Default HTTPRequent configuration values.
+    """
+
+    def __init__(self, api_config, namespace_http_config=None):
```

Comment:
again, style nits ( hehe, i'm *full* of them ).  namespace_http_config is humongous ;) 

Also, is there any reason we need two configs rather than making the http_config a value within the api_config ( which would then just be "conf" since there's only one?  Having them nested within one would probably make deployment/config-management a bit simpler.